File: Warehousing.Repo/Dtos/OrderDto.cs
namespace Warehousing.Repo.Dtos
{
    public class OrderDto
    {
        public int? Id { get; set; }
        public DateTime OrderDate { get; set; }
        public decimal TotalAmount { get; set; }

        //FK
        public OrderTypeDto? OrderType { get; set; }
        public int? OrderTypeId { get; set; }
        public CustomerDto? Customer { get; set; }
        public int? CustomerId { get; set; }
        public SupplierDto? Supplier { get; set; }
        public int? SupplierId { get; set; }
        public StatusDto? Status { get; set; }
        public int? StatusId { get; set; }

        public ICollection<OrderItemDto> Items { get; set; } = new List<OrderItemDto>();
    }
}

File: Warehousing.UI/src/app/admin/models/OrderDto.ts
import { Customer } from "./customer";
import { OrderItemDto } from "./OrderItemDto";
import { OrderTypeDto } from "./OrderTypeDto";
import { Status } from "./status";
import { Supplier } from "./supplier";

export interface OrderDto {
    id: number;
    orderDate: string;
    totalAmount: number;
    orderType: OrderTypeDto | null;
    orderTypeId: number | null;
    customer: Customer | null;
    customerId: number | null;
    supplier: Supplier | null;
    supplierId: number | null;
    status: Status | null;
    statusId: number | null;
    items: OrderItemDto[];
    statusColor: string;
}

export interface OrderPagination {
    orders: OrderDto[],
    totals: number
}

export interface OrderFilters {
    pageIndex: number;
    pageSize: number;
    searchTerm: string | null;
    orderDate: string | null;
    orderTypeId: number | null;
    customerId: number | null;
    supplierId: number | null;
    statusId: number | null;
    dateFrom: string | null;
    dateTo: string | null;
    minAmount: number | null;
    maxAmount: number | null;
}